app/assets/javascripts/discourse/app/lib/plugin-api.gjs

```diff
@@ -1803,49 +1804,95 @@ class PluginApi {
   /**
    * Allows adding icons to the category-link html
    *
-   * ```
+   * ```javascript
    * api.addCategoryLinkIcon((category) => {
-   *  if (category.someProperty) {
-        return "eye"
-      }
+   *   if (category.someProperty) {
+   *     return "eye"
+   *   }
    * });
    * ```
-   *
    **/
   addCategoryLinkIcon(renderer) {
     addExtraIconRenderer(renderer);
   }
+
   /**
-   * Adds a widget or a component to the header-icon ul.
+   * Allows for manipulation of the header icons. This includes, adding, removing, or modifying the order of icons.
    *
-   * If adding a widget it must already be created. You can create new widgets
-   * in a theme or plugin via an initializer prior to calling this function.
+   * Only the passing of components is supported, and by default the icons are added to the right of exisiting icons.
    *
+   * Example: Add the chat icon to the header icons before the search icon
    * ```
-   * api.addToHeaderIcons(
-   *  createWidget("some-widget")
+   * api.headerIcons.add(
+   *  "chat",
+   *  ChatIconComponent,
+   *  { before: "search" }
+   * )
```

Comment:
Wondering if we should make `{before: "search"}` the default for header icons? Seems like that would be make it a better match for the old `addHeaderIcon` API?

In terms of implementation, maybe `new DAG()` could accept a `defaultPosition` argument? 🤔